src/main/java/uk/gov/hmcts/darts/courthouse/controller/CourthousesController.java

```diff
@@ -39,15 +54,32 @@ public ResponseEntity<Void> courthousesCourthouseIdDelete(
     }
 
     @Override
-    public ResponseEntity<ExtendedCourthouse> courthousesCourthouseIdGet(
+    @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
+    @Authorisation(contextId = ANY_ENTITY_ID, globalAccessSecurityRoles = ADMIN)
+    public ResponseEntity<AdminCourthouse> adminCourthousesCourthouseIdGet(
         @Parameter(name = "courthouse_id", description = "", required = true, in = ParameterIn.PATH) @PathVariable("courthouse_id") Integer courthouseId
     ) {
         try {
-            CourthouseEntity courtHouseEntity = courthouseService.getCourtHouseById(
-                courthouseId);
-            ExtendedCourthouse responseEntity = mapper.mapFromEntityToExtendedCourthouse(courtHouseEntity);
+            CourthouseEntity courtHouseEntity = courthouseService.getCourtHouseById(courthouseId);
+            Set<SecurityGroupEntity> secGrps = courtHouseEntity.getSecurityGroups();
+            List<Integer> secGrpIds = new ArrayList<>();
+            if (secGrps != null && !secGrps.isEmpty()) {
+                secGrpIds = secGrps.stream().map(SecurityGroupEntity::getId).collect(Collectors.toList());
+            }
+
+            AdminCourthouse responseEntity = adminMapper.mapFromEntityToAdminCourthouse(courtHouseEntity);
+            responseEntity.setSecurityGroupIds(secGrpIds);
+
+            RegionEntity region = courtHouseEntity.getRegion();
+
+            Integer regionId = null;
+            if (region != null) {
+                regionId = region.getId();
+            }
+            responseEntity.setRegionId(regionId);
```

Comment:
normally we would have all this logic in the service, instead of the controller

Reply:
Code moved to CourthouseServiceImpl::getAdminCourtHouseById(Integer id)